homeassistant/components/device_tracker/proximity.py

```diff
@@ -0,0 +1,279 @@
+        distance_from_zone = round(distance(proximity_latitude, proximity_longitude, new_state.attributes['latitude'], new_state.attributes['longitude']) / 1000, 1)
+        _LOGGER.info('%s: distance from zone is: %s km', entity_name, distance_from_zone)
+
+        """========================================================"""  
+        #compare distance with other devices
+        #default behaviour
+        device_is_closest_to_zone = True
+
+        for device in proximity_devices:
+            #ignore the device we're working on
+            if device == entity:
+                continue
+
+            #get the device state
+            device_state = hass.states.get(device)
+            if device_state in override_zones:
+                _LOGGER.debug('%s: no need to compare with %s - device is in override zone', entity_name, device)
+                continue
+
+            #check that the distance from the proximity zone can be calculated
+            if not('latitude' in device_state.attributes):
+                _LOGGER.debug('%s: cannot compare with %s - no location attributes', entity_name, device)
+                continue
+
+            #calculate the distance from the proximity zone for the compare device
+            compare_distance_from_zone = round(distance(proximity_latitude, proximity_longitude, device_state.attributes['latitude'], device_state.attributes['longitude'])/1000 ,1)
+            _LOGGER.debug('%s: compare device %s: co-ordintes: LAT %s: LONG: %s', entity_name, device, device_state.attributes['latitude'], device_state.attributes['longitude'])
+
+            #compare the distances from the proximity zone
+            if distance_from_zone < compare_distance_from_zone:
+                _LOGGER.debug('%s: closer than %s: %s compared with %s', entity_name, device, distance_from_zone, compare_distance_from_zone)
+            elif distance_from_zone > compare_distance_from_zone:
+                device_is_closest_to_zone = False
+                _LOGGER.debug('%s: further away than %s: %s compared with %s', entity_name, device, distance_from_zone, compare_distance_from_zone)
+            else:
+                device_is_closest_to_zone = False
+                _LOGGER.debug('%s: same distance as %s: %s compared with %s', entity_name, device, distance_from_zone, compare_distance_from_zone)
+
+        """========================================================"""
+        #if the device is not the closest to the proximity zone
+        if device_is_closest_to_zone == False:
+            _LOGGER.info('%s: device is not closest to zone', entity_name)
+            return
+
+        """========================================================"""
+        #calculate direction of travel
+        #stop if we cannot calculate the direction of travel (i.e. we don't have a previous state and a current LAT and LONG)
+        if old_state == None or new_state.attributes['latitude'] == None or not('latitude' in old_state.attributes):
```

Comment:
```python
if old_state is None or …
```

The test `new_state.attributes['latitude'] == None` will fail, the safe guard is the check after it. You can basically remove this specific test.